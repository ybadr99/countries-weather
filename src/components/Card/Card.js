import React from 'react';
import './Card.scss';
import { useNavigate } from 'react-router-dom';

const Card = ({ country }) => {
  const {
    name, population, region, capital, flag,
  } = country;

  const navigator = useNavigate();

  const formattedPopulation = new Intl.NumberFormat('tr-TR').format(population);

  return (
    <div className="card" onClick={() => navigator(`/${name}`)} aria-hidden="true">
      <img src={flag} alt="Flag" />
      <ol>
        <h2>{name}</h2>
        <li>
          Population:
          <span>{formattedPopulation}</span>
        </li>
        <li>
          Region:
          <span>{region}</span>
        </li>
        <li>
          Capital:
          <span>{capital}</span>
        </li>
      </ol>
    </div>
  );
};

export default Card;
