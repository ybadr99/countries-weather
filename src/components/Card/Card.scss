@import "../../style/variables.scss";

.card {
  display: flex;
  flex-direction: column;
  background-color: $dark-blue;
  border-bottom-left-radius: 0.2rem;
  border-bottom-right-radius: 0.2rem;
  max-width: 400px;
  cursor: pointer;

  img {
    object-fit: cover;
    border-top-left-radius: 0.2rem;
    border-top-right-radius: 0.2rem;
    width: 100%;
    height: 10rem;
  }

  ol {
    padding: 1rem;
    color: $white;

    h2 {
      margin-bottom: 1rem;
    }

    li {
      font-weight: 600;

      span {
        font-weight: 300;
      }
    }
  }
}