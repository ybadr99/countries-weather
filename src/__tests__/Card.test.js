import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Card from '../components/Card/Card';

const country = {
  name: 'Afghanistan',
  population: 40218234,
  region: 'Asia',
  capital: 'Kabul',
  flag: 'https://upload.wikimedia.org/wikipedia/commons/5/5c/Flag_of_the_Taliban.svg',
};

describe('Card component', () => {
  it('renders the card correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Card country={country} />
      </MemoryRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
