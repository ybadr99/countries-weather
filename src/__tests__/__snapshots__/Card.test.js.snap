// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the card correctly 1`] = `
<div
  aria-hidden="true"
  class="card"
>
  <img
    alt="Flag"
    src="https://upload.wikimedia.org/wikipedia/commons/5/5c/Flag_of_the_Taliban.svg"
  />
  <ol>
    <h2>
      Afghanistan
    </h2>
    <li>
      Population:
      <span>
        40.218.234
      </span>
    </li>
    <li>
      Region:
      <span>
        Asia
      </span>
    </li>
    <li>
      Capital:
      <span>
        Kabul
      </span>
    </li>
  </ol>
</div>
`;
